HDFDataset.py

```diff
@@ -257,10 +272,14 @@ def get_tag(self, sorted_seq_idx):
     return self._get_tag_by_real_idx(ids)
 
   def get_all_tags(self):
-    return list(map(self._decode, self._tags))
+    tags = []
+    for h5_file in self.h5_files:
+      tags += h5_file["seqTags"][...].tolist()
+
+    return list(map(self._decode, tags))
 
   def get_total_num_seqs(self):
-    return len(self._tags)
+    return self._num_seqs
```

Comment:
This is supposed to return the total number of seqs (without partition epoch or so). `_num_seqs` (if it is set) is for one specific epoch. Maybe `len(self._seq_index)` is right here? Not sure.

Reply:
Sadly, in HDFDataset _num_seqs is actually the total number. It's added up in add_file() and never updated. But to use the same name as in _num_seqs in CachedDataset2 is definitively confusing, I agree.

Reply:
Or worse, it's an attribute of Dataset even, but used differently in CachedDataset and CachedDataset2.

Reply:
Oh, then `HDFDataset` might actually be an example of a dataset where you cannot rely on `num_seqs` (it can be different, depending e.g. if you use a predefined seq list, or use epoch split, or sth like that). Although, I don't know. `_num_seqs` is supposed to be internal anyway, and as long as `num_seqs` is right, it should not matter from the outside.